mono/metadata/threadpool-ms-io.c

```diff
@@ -311,14 +312,19 @@ wait_callback (gint fd, gint events, gpointer user_data)
 				mono_threadpool_ms_enqueue_work_item (((MonoObject*) sockares)->vtable->domain, (MonoObject*) sockares);
 		}
 
-		mono_g_hash_table_replace (states, GINT_TO_POINTER (fd), list);
-
-		events = get_events (list);
-
-		mono_trace (G_LOG_LEVEL_DEBUG, MONO_TRACE_IO_THREADPOOL, "io threadpool: res fd %3d, events = %2s | %2s",
-			fd, (events & EVENT_IN) ? "RD" : "..", (events & EVENT_OUT) ? "WR" : "..");
-
-		threadpool_io->backend.register_fd (fd, events, FALSE);
+		remove_fd = (events & EVENT_ERR) == EVENT_ERR;
+		if (!remove_fd) {
+			mono_g_hash_table_replace (states, GINT_TO_POINTER (fd), list);
+			events = get_events (list);
+
+			mono_trace (G_LOG_LEVEL_DEBUG, MONO_TRACE_IO_THREADPOOL, "io threadpool: res fd %3d, events = %2s | %2s | %2s",
+				    fd, (events & EVENT_IN) ? "RD" : "..", (events & EVENT_OUT) ? "WR" : "..", (events & EVENT_ERR) ? "ERR" : "...");
+			threadpool_io->backend.register_fd (fd, events, FALSE);
+		} else {
+			mono_trace (G_LOG_LEVEL_DEBUG, MONO_TRACE_IO_THREADPOOL, "io threadpool: removing fd %d because of error", fd);
```

Comment:
Can you just replace the "removing fd %d because of error" by "err %d", so we can keep it 3 letters (like the other ones), so when debugging, it's easy to find fd in the output. Otherwise, it's good to merge. Thank you!